test/new-e2e/tests/agent-metric-logs/linux-log-agent/01_file_tailing_test.go

```diff
@@ -0,0 +1,161 @@
+		e2e.WithAgentParams(
+			agentparams.WithLogs(),
+			agentparams.WithIntegration("custom_logs.d", config)))
+
+}
+
+// TestE2EVMFakeintakeSuite runs the E2E test suite for the log agent with a VM and fake intake.
+func TestE2EVMFakeintakeSuite(t *testing.T) {
+	e2e.Run(t, &vmFakeintakeSuite{}, logsExampleStackDef(), params.WithDevMode())
+}
+
+func (s *vmFakeintakeSuite) TestLinuxLogTailing() {
+	// Clean up once test is finished running
+	s.cleanUp()
+	defer s.cleanUp()
+
+	// Flush server and reset aggregators
+	s.Env().Fakeintake.FlushServerAndResetAggregators()
+	defer s.Env().Fakeintake.FlushServerAndResetAggregators()
+
+	// Run test cases
+	s.T().Run("LogCollection", func(t *testing.T) {
+		s.LogCollection()
+	})
+
+	s.T().Run("LogPermission", func(t *testing.T) {
+		s.LogPermission()
+	})
+
+	s.T().Run("LogRotation", func(t *testing.T) {
+		s.LogRotation()
+	})
+}
+
+func (s *vmFakeintakeSuite) LogCollection() {
+	t := s.T()
+	fakeintake := s.Env().Fakeintake
+
+	// Create a new log file
+	s.Env().VM.Execute("sudo touch /var/log/hello-world.log")
+
+	// Part 1: Ensure no logs are present in fakeintake
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		logs, err := fakeintake.FilterLogs("hello")
+		require.NoError(t, err, "Unable to filter logs by the service 'hello'.")
+		require.Empty(t, logs, "Logs were found when none were expected.")
+
+		// If logs are found, print their content for debugging
+		if len(logs) != 0 {
+			cat, _ := s.Env().VM.ExecuteWithError("cat /var/log/hello-world.log")
+			t.Logf("Logs detected when none were expected: %v", cat)
+			require.Empty(t, logs, "Logs were found when none were expected.")
+		}
+	}, 5*time.Minute, 10*time.Second)
+
+	// Part 2: Adjust permissions of new log file
+	_, err := s.Env().VM.ExecuteWithError("sudo chmod 777 /var/log/hello-world.log")
+	require.NoError(t, err, "Unable to adjust permissions for the log file '/var/log/hello-world.log'.")
+
+	// Generate log
+	generateLog(s, t, "hello-world")
+
+	// Part 3: Assert that logs are found in intake after generation
+	checkLogs(s, "hello", "hello-world")
+}
+
+func (s *vmFakeintakeSuite) LogPermission() {
+	t := s.T()
+
+	// Part 4: Block permission and check the Agent status
+	s.Env().VM.Execute("sudo chmod 000 /var/log/hello-world.log")
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		// Check the Agent status
+		statusOutput, err := s.Env().VM.ExecuteWithError("sudo datadog-agent status | grep -A 10 'custom_logs'")
+		require.NoError(t, err, "Issue running agent status: %s", err)
+
+		if strings.Contains(statusOutput, "Status: OK") {
+			require.Fail(t, "log file is unexpectedly accessible")
+		}
+
+		require.Contains(t, statusOutput, "denied", "Log file is correctly inaccessible")
+	}, 3*time.Minute, 10*time.Second)
+
+	// Part 5: Restore permissions
+	s.Env().VM.Execute("sudo chmod 777 /var/log/hello-world.log")
+
+	// Part 6: Restart the agent, generate new logs
+	s.Env().VM.Execute("sudo service datadog-agent restart")
```

Comment:
This might be way way back when the agent was in 7.10x or something but that used to be the behavior where users have to manually restart the agent when the permission changes 😅 . I just did a quick test and it looks like the agent now will rescan just fine when the permission changes so I'll get rid of this line

Reply:
Might be a flaky issue, but it looks like Eventhogh the agent should rescan, the test fails to get new logs unless the agent is restarted. i can bring it up to OH next week

Reply:
Interesting, is this an expected feature ?

Reply:
After this you could add a call to wait for the agent to be ready, there should be an helper on the agent client